internal/escher_api/model/cd_serverless_rule.go

```diff
@@ -0,0 +1,409 @@
+// Code generated by go-swagger; DO NOT EDIT.
+
+package model
+
+// This file was generated by the swagger tool.
+// Editing this file might prove futile when you re-run the swagger generate command
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"strconv"
+
+	"github.com/go-openapi/errors"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/go-openapi/validate"
+)
+
+// CdServerlessRule A rule that states what serverless function are allowed and where.
+//
+// swagger:model CdServerlessRule
+type CdServerlessRule struct {
+
+	// action
+	// Required: true
+	Action *ServerlessRuleAction `json:"action"`
+
+	// group name
+	GroupName string `json:"groupName,omitempty"`
+
+	// id
+	// Format: uuid
+	ID strfmt.UUID `json:"id,omitempty"`
+
+	// name
+	// Required: true
+	Name *string `json:"name"`
+
+	ruleField ServerlessRuleType
```

Comment:
why serverlessRuleType is not a pointer as in serverlessRuleStatus?

Reply:
this is how the auto generator generated it